__tests__/utils/permissions.test.js

```diff
@@ -48,27 +56,21 @@ describe('requestExternalStoragePermission shpuld run correctly', () => {
     }));
   });
   it('requestExternalStoragePermission granted should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('granted');
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue(mockPermission('granted'));
     const checkPermission = await requestExternalStoragePermission();
     expect(mock.check).toHaveBeenCalled();
     expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
   });
   it('requestExternalStoragePermission unavailable should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('unavailable');
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue(mockPermission('unavailable'));
     const checkPermission = await requestExternalStoragePermission();
     expect(mock.check).toHaveBeenCalled();
-    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+    expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionUnavailable);
   });
   it('requestExternalStoragePermission denied should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('denied');
-    const checkPermission = await requestExternalStoragePermission();
-    expect(mock.check).toHaveBeenCalled();
-    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
-  });
-  it('requestExternalStoragePermission default should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('default');
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue(mockPermission('denied'));
     const checkPermission = await requestExternalStoragePermission();
     expect(mock.check).toHaveBeenCalled();
-    expect(checkPermission.message).toEqual('');
+    expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionDenied);
   });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a mistake in your test cases. You are spying on `mock.requestMultiple` but then checking if `mock.check` has been called. This will lead to false positives in your tests as `mock.check` is not being called anymore. You should replace `expect(mock.check).toHaveBeenCalled();` with `expect(mock.requestMultiple).toHaveBeenCalled();`.

```diff
- expect(mock.check).toHaveBeenCalled();
+ expect(mock.requestMultiple).toHaveBeenCalled();
```
Also, there's no newline at the end of the file. It's a good practice to always end a file with a newline.

```diff
   });
\ No newline at end of file
+    });
+    
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
this is only new line issue